### PR TITLE
add Google OAuth Login

### PR DESCRIPTION
**Pull request by amritanshusikdar**
# Added Google OAuth login Feature.
In order to test it, `clientId` has to be generated from Google's developer console and pasted in the .env file under:
> /front

### CODE REVIEW

File: front/src/components/header/subcomponents/LoginModal.tsx [hunk 1]
```diff
@@ -245,48 +300,49 @@ const LoginModal = ({ disclosureProps, setIsLoggedIn, unregisteredClearFollowedS
...
-                    fontSize="0.88rem"
-                    onClick={() => setFormOption((formOption + 1) % 2)}
-                    pt="0.75rem"
-                    variant="link"
-                  >
-                    {(formOption === 0 && 'Forgot Password?') ||
-                      (formOption === 1 && (
-                        <>
-                          <Box as={TiArrowBack} size="18px" />
-                          Go back
```
Comment (trybick) [inline front/src/components/header/subcomponents/LoginModal.tsx:272]: I think there might have been a merge conflict here - this version of the Back button with `TiArrowBack` is the newer version and should be included, but below the version of `< Back to Login` is included which is older

File: front/src/components/header/subcomponents/LoginModal.tsx [hunk 2]
```diff
@@ -119,6 +122,58 @@ const LoginModal = ({ disclosureProps, setIsLoggedIn, unregisteredClearFollowedS
       });
   };
 
+  //  Handles OAuth login from Google
+  type Responses = GoogleLoginResponse | GoogleLoginResponseOffline;
+  const handleGoogleLogin = {
```
Comment (trybick) [inline front/src/components/header/subcomponents/LoginModal.tsx]: Could we move this function and type into a new file? - something like `utils/google0Auth.ts`

File: front/src/components/header/subcomponents/LoginModal.tsx (hunk 1 above)
Comment (amritanshusikdar) [reply front/src/components/header/subcomponents/LoginModal.tsx:272]: I updated my forked repository and cloned the updated one, but still it didn't reflect the changes. Weird. Alright, I'm updating the code right now.

File: front/src/components/header/subcomponents/LoginModal.tsx (hunk 2 above)
Comment (amritanshusikdar) [reply front/src/components/header/subcomponents/LoginModal.tsx]: Sure

**Review by amritanshusikdar — COMMENTED** [review]
All the modifications have been done and the potential merge conflicts have been fixed.

**Pull request merged**